lib/train/platforms/detect/helpers/os_common.rb

```diff
@@ -34,16 +34,18 @@ def unix_file_exist?(path)
     end
 
     def command_output(cmd)
-      res = @backend.run_command(cmd).stdout
+      res = @backend.run_command(cmd)
+      output = res.stdout
+      output = res.stderr if output == ""  && res.exit_status != 0
       # When you try to execute command using ssh connction as root user and you have provided ssh user identity file
       # it gives standard output to login as authorised user other than root. To show this standard ouput as an error
       # to user we are matching the string of stdout and raising the error here so that user gets exact information.
-      if @backend.class.to_s == "Train::Transports::SSH::Connection" && res =~ /Please login as the user/
-        raise Train::UserError, "SSH failed: #{res}"
+      if @backend.class.to_s == "Train::Transports::SSH::Connection" && (output =~ /Please login as the user/ || output =~ /WARNING: Your password has expired.\nPassword change required but no TTY available.\n/)
+        raise Train::UserError, "SSH failed: #{output}"
```

Comment:
this might not be an optimal solution, any suggestion would be appreciated. Thanks!

Reply:
Maybe this would be more readable as 

```
if @backend.class.to_s == "Train::Transports::SSH::Connection" && (tdout =~ /Please login as the user/ 
    raise Train::UserError, "SSH failed: #{stdout}"
end
if @backend.class.to_s == "Train::Transports::SSH::Connection" && stderr =~ /WARNING: Your password has expired/)
    raise Train::UserError, "SSH failed: #{stderr}"
end
```

There's duplication between the two `if` statements but I think it's OK because it's expressing two separate failure cases.